### DISCUSSION TITLE
مشکل اتصال نود به پنل مرزبان نصب شده روی سرور ایران

### DISCUSSION THREAD

**Discussion by asbsiah** [Gozargah/Marzban#1318] [Q&A]
سلام وقت بخیر 
سرور مرزبان رو با دردسر های زیاد به صورت صحیح روی سرور ایران نصب کردم و پنل مشکلی وجود ندارد
اما با اتصال نود به پنل مشکلی وجود دارد 
در لاگ پنل مرزبان 

marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     Connecting to "ge" node
marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     Connecting to "ge" node
marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     Connecting to "ge" node
marzban-1  | INFO:     5.125.53.12:21832 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21833 - "GET /api/node/settings HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     Connecting to "ge" node
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     Connecting to "ge" node
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     Unable to connect to "ge" node
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/system HTTP/1.1" 200 OK
marzban-1  | INFO:     5.125.53.12:21834 - "GET /api/nodes HTTP/1.1" 200 OK

در پنل مرزبان 
result expired

و سرویس داکر نود به این صورت میباشد 

services:
  marzban-node-ge1:
    # build: .
    image: gozargah/marzban-node:latest
    restart: always
    network_mode: host

    environment:
      SERVICE_PORT: 2024
      XRAY_API_PORT: 2025
      SSL_CLIENT_CERT_FILE: "/var/lib/marzban-node/ssl_client_cert_ge1.pem"
      XRAY_EXECUTABLE_PATH: "/var/lib/marzban/xray-core/xray"

    volumes:
      - /var/lib/marzban-node:/var/lib/marzban-node
      - /var/lib/marzban:/var/lib/marzban

دقیقا به همین روش مرسوم پنل روی سرور خارج و با همین سرویس نود مورد نظر متصل شده است اما وقتی پنل روی ایران نصب شود این مشکلی ایجاد میشود و مدام در حالت Connecting to "ge" node قرار میگیرد
![afafff](https://github.com/user-attachments/assets/563910e0-44ce-4469-937d-2580f4fac90b)

**Comment by TahaJalali** [comment]
برای من هم این مشکل وجود داره

**Reply by asbsiah** [reply]
سلام عزیز وقت بخیر 
در نهایت روش نصب پنل مرزبان رو به راحتی هر چه تمام تر تونستم حل کنم توی یک دیسکا جدید میزارش اما در نتیجه برای اتصال نود به پنل ایران به این نتیجه رسیدم اینجا میگم که دوستان متضرر نشن : 

در استفاده از داکر بر روی سرور ها و آی پی های ایران محدودیت هایی وجود دارد شما با دور زدن محدودیت ها روی سرور های ایران در نهایت موفق به نصب پنل می شوید اما برای اتصال یک نود خارج به سرور پنل ایران مرزبان مشکل حل نشدنی تا اینجای کار برقرار است 
مشکل زمانی پیش میاد که حالا دو تا سرویس داکر برای ارتباط با هم دچار مشکل میشن و اون اتصال خارجی پنل مرزبان که روی سرور ایران نصب شده با نود خودش در خارج از طریق داکر هست یعنی :

داکر با داکر نمیتونه اتصال برقرار بکنه چون سرور داکر شما در پنل مرزبان داخلی نصب هست و به خاطر محدودیت ها از اتصال این ارتباط توسط داکر جلوگیری میشه 

ممنون میشم از برنامه نویسای زحمتکش مرزبان که اگر راه حلی برای اتصال داکر به داکر با در نظر گرفتن سرور پنل ایران ارائه بکنند 

ممنون از شما ها
